src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingManagementController.java

```diff
@@ -200,6 +189,58 @@ public ResponseEntity hearingCompletion(@PathVariable("id") Long hearingId) {
         return hearingManagementService.hearingCompletion(hearingId);
     }
 
+    /**
+     * get list of cases either by caseRefId OR CaseRefId/caseStatus.
+     * @param ccdCaseRefs list of case Ref
+     * @param status optional Status
+     * @return Hearing
+     */
+    @Transactional
+    @GetMapping(value = {"/hearings"},
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "Get hearings for list of cases")
+    @ApiResponses({
+        @ApiResponse(code = 200, message = "Success (with content)"),
+        @ApiResponse(code = 400,
+            message = "One or more of the following reasons:"
+                + "\n1) " + ValidationError.INVALID_HEARING_REQUEST_DETAILS
+        )
+    })
+    public List<GetHearingsResponse> getHearingsForListOfCases(@RequestParam List<String> ccdCaseRefs,
+                                                         @RequestParam(required = false)
+                                                         String status) {
+        List<GetHearingsResponse> hearingsResponseList = new ArrayList<>();
+        if (ccdCaseRefs.isEmpty()) {
+            throw new BadRequestException(CASE_REF_EMPTY);
+        }
+        for (String ccdCaseRef : ccdCaseRefs) {
+            GetHearingsResponse hearingsResponse = getHearingsResponse(ccdCaseRef, status);
+            if (hearingsResponse.getCaseHearings().size() != 0) {
+                hearingsResponseList.add(hearingsResponse);
+            }
+        }
+        return hearingsResponseList.isEmpty() ? null : hearingsResponseList;
+
+    }
+
+    private GetHearingsResponse getHearingsResponse(String ccdCaseRef, String status) {
+        List<String> filteredRoleAssignments =
+            accessControlService.verifyCaseAccess(ccdCaseRef, Lists.newArrayList(
+                HEARING_VIEWER,
+                LISTED_HEARING_VIEWER));
+
+        if (hasOnlyListedHearingViewerRoles(filteredRoleAssignments)) {
+            if ((status == null || HearingStatus.LISTED.name().equals(status))) {
+                status = HearingStatus.LISTED.name();
```

Comment:
What's the point of this assignment?

Reply:
listed-hearing-viewer is allowed to view only hearings with LISTED status